client/src/partials/reports_proposed/debitor_group_report/debitor_group_report.js

```diff
@@ -0,0 +1,74 @@
+var DebtorGroupReportController = function ($http, $translate, validate, reportConfigService, messenger) {
+
+  var vm = this,
+  generatedDocumentPath = null,
+  serverUtilityPath = '/report/build/debitor_group_report',
+  configuration = reportConfigService.configuration,
+  dependencies = {};
+
+  dependencies.debitorGroup = {
+      query : {
+        identifier : 'uuid',
+        tables : {
+          'debitor_group' : {
+            columns : ['uuid', 'name', 'account_id', 'is_convention']
+          }
+        },
+        where : ['debitor_group.is_convention=1']
+      }
+  };
+
+  vm.generate_doc = $translate.instant('DEBITOR_GROUP_REPORT.GENERATE_DOC');
```

Comment:
I'm almost positive that it only updates translation because the controller is re-run.  Although, [the documentation](https://angular-translate.github.io/docs/#/api/pascalprecht.translate.$translate) does not make it clear if $translate.instant() is scheduled to update on language change.  However, if you read the [source code](https://github.com/angular-translate/angular-translate/blob/920a9fe/src/service/translate.js#L1935), it looks like the instant method does not schedule listeners to be run multiple times.

You already have logic in the view...at this very spot.  But if you do not want to put translation logic in the view, just do
```js
vm.message = vm.loading ? 'DEBITOR_GROUP_REPORT.GENERATE_DOC' : 'DEBITOR_GROUP_REPORT.LOADING';
```

and update accordingly.